specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/preview/2020-05-01-preview/machineLearningServices.json

```diff
@@ -0,0 +1,5214 @@
+              }
+            },
+            "models": {
+              "description": "The list of models.",
+              "type": "array",
+              "items": {
+                "$ref": "#/definitions/Model"
+              }
+            },
+            "environmentImageRequest": {
+              "description": "The Environment, models and assets used for inferencing.",
+              "allOf": [
+                {
+                  "$ref": "#/definitions/EnvironmentImageRequest"
+                }
+              ]
+            },
+            "vnetConfiguration": {
+              "description": "The virtual network configuration.",
+              "allOf": [
+                {
+                  "$ref": "#/definitions/VnetConfiguration"
+                }
+              ]
+            },
+            "encryptionProperties": {
+              "description": "The encryption properties.",
+              "allOf": [
+                {
+                  "$ref": "#/definitions/EncryptionProperties"
+                }
+              ]
+            }
+          }
+        }
+      ],
+      "x-ms-discriminator-value": "ACI"
+    },
+    "ContainerResourceRequirements": {
+      "description": "The resource requirements for the container (cpu and memory).",
+      "type": "object",
+      "properties": {
+        "cpu": {
+          "format": "double",
+          "description": "The number of CPU cores on the container.",
+          "type": "number",
+          "example": 4
+        },
+        "memoryInGB": {
+          "format": "double",
+          "description": "The amount of memory on the container in GB.",
+          "type": "number",
+          "example": 64
+        },
+        "gpu": {
+          "format": "int32",
+          "description": "The number of GPU cores in the container.",
+          "type": "integer"
+        },
+        "fpga": {
+          "format": "int32",
+          "description": "The number of FPGA PCIE devices exposed to the container. Must be multiple of 2.",
+          "type": "integer"
+        }
+      }
+    },
+    "ModelDataCollection": {
+      "description": "The Model data collection properties.",
+      "type": "object",
+      "properties": {
+        "eventHubEnabled": {
+          "description": "Option for enabling/disabling Event Hub.",
+          "type": "boolean"
+        },
+        "storageEnabled": {
+          "description": "Option for enabling/disabling storage.",
+          "type": "boolean"
+        }
+      }
+    },
+    "VnetConfiguration": {
+      "type": "object",
+      "properties": {
+        "vnetName": {
+          "description": "The name of the virtual network.",
+          "type": "string"
+        },
+        "subnetName": {
+          "description": "The name of the virtual network subnet.",
+          "type": "string"
+        }
+      }
+    },
+    "EncryptionProperties": {
+      "type": "object",
+      "properties": {
+        "vaultBaseUrl": {
```

Comment:
these should all be marked as required in this definition right?  I.e. if a user provides encryptionProperties they need to provide all 3 of these values.  Please make sure you are marking required properties as such

Reply:
made changes